tests/functional/test_colorama.py

```diff
@@ -0,0 +1,222 @@
+        self.assertEqual(Fore.LIGHTWHITE_EX, '\033[97m')
+
+    def test_back_attributes(self):
+        self.assertEqual(Back.BLACK, '\033[40m')
+        self.assertEqual(Back.RED, '\033[41m')
+        self.assertEqual(Back.GREEN, '\033[42m')
+        self.assertEqual(Back.YELLOW, '\033[43m')
+        self.assertEqual(Back.BLUE, '\033[44m')
+        self.assertEqual(Back.MAGENTA, '\033[45m')
+        self.assertEqual(Back.CYAN, '\033[46m')
+        self.assertEqual(Back.WHITE, '\033[47m')
+        self.assertEqual(Back.RESET, '\033[49m')
+
+        # Check the light, extended versions.
+        self.assertEqual(Back.LIGHTBLACK_EX, '\033[100m')
+        self.assertEqual(Back.LIGHTRED_EX, '\033[101m')
+        self.assertEqual(Back.LIGHTGREEN_EX, '\033[102m')
+        self.assertEqual(Back.LIGHTYELLOW_EX, '\033[103m')
+        self.assertEqual(Back.LIGHTBLUE_EX, '\033[104m')
+        self.assertEqual(Back.LIGHTMAGENTA_EX, '\033[105m')
+        self.assertEqual(Back.LIGHTCYAN_EX, '\033[106m')
+        self.assertEqual(Back.LIGHTWHITE_EX, '\033[107m')
+
+    @contextmanager
+    def colorama_text(self, tty=True):
+        with capture_output() as captured:
+            captured.stdout.isatty = lambda: tty
+            with colorama.colorama_text(**self._COLORAMA_KWARGS):
+                yield captured
+
+    def test_colorama_auto_resets(self):
+        with self.colorama_text() as captured:
+             content = Fore.RED + 'foo'
+             sys.stdout.write(content)
+             sys.stdout.write('bar')
+        # Since auto-reset is enabled each call to write should end with the
+        # reset code. And foo should start with the red foreground code. This
+        # also depends on the strip=False behavior to pass on windows.
+        self.assertEqual(captured.stdout.getvalue(),
+                         '%sfoo%sbar%s' % (self._ANSI_FORE_RED,
+                                           self._ANSI_RESET_ALL,
+                                           self._ANSI_RESET_ALL))
+
+    def test_colorama_does_not_strip(self):
+        # Strip is set to False which means that it doesn't remove ansi codes
+        # from the output if its a pipe.
+        with self.colorama_text() as captured:
+            content = Fore.BLUE + 'foo'
+            sys.stdout.write(content)
+            self.assertEqual(captured.stdout.getvalue(),
+                             '%sfoo%s' % (self._ANSI_FORE_BLUE,
+                                          self._ANSI_RESET_ALL))
+
+    def test_colorama_does_not_strip_non_tty(self):
+        with self.colorama_text(tty=False) as captured:
+            content = Fore.BLUE + 'foo'
+            sys.stdout.write(content)
+            self.assertEqual(captured.stdout.getvalue(),
+                             '%sfoo%s' % (self._ANSI_FORE_BLUE,
+                                          self._ANSI_RESET_ALL))
+
+
+
+@unittest.skipUnless(sys.platform.startswith("win"), "requires Windows")
+class WinTermTest(unittest.TestCase):
```

Comment:
I don't think these windows tests are particularly useful for us.  This seems to be testing the internals of how their winterm is implemented, and we don't directly use any of those classes in the CLI itself.

Assuming the `TestPosix` behavior also applies to windows (if you pipe the output you still get colorized output), I'd prefer to see similar high level tests for windows.  E.g. "test_colorama_does_not_strip_non_tty" etc., even if we have to restrict these tests to only run if we're on powershell (or cmd or whatever).